servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -250,8 +251,14 @@ public StreamingHttpClient buildStreaming() {
     }
 
     private static <U, R> StreamingHttpClient buildStreaming(final HttpClientBuildContext<U, R> ctx) {
-        final ReadOnlyHttpClientConfig roConfig = ctx.httpConfig().asReadOnly();
         final HttpExecutionContext executionContext = ctx.builder.executionContextBuilder.build();
+        // XXX This should probably be somewhere else
+        if (noOffloadsStrategy() == executionContext.executionStrategy()) {
```

Comment:
I was originally assuming `HttpExecutionStrategy` would have something like `isCloseOffloaded`, which would solve this issue but I think you mentioned this would complicate merging/combination. What is the expected behavior if the user says `noOffloadsStrategy()` but offloads `asyncCloseOffload`?

Reply:
We had an offline discussion with Mike about having this as a builder option or promoting to `ExecutionStrategy` interface as an additional flag. Rational was that almost everything extends `AsyncCloseable` interface, including `ConnectionFactory`, `ConnectionAcceptor`, HTTP filters, etc. All may apply blocking code for `closeAsync` or `onClose` processing and should be able to influence its behavior. It's also useful for setting as an `HttpExecutionStrategy` at the builder level, because `onClose()` can be used outside of the filters, and will eliminate question "What is the expected behavior if the user says `noOffloadsStrategy()` but offloads asyncCloseOffload?".

@bondolo have you had a chance to consider this approach?

Reply:
I think it is worth considering as control of `close()` offloading is a common enough behaviour. I haven't had time to try implementing it as an `ExecutionStrategy` flag.

Reply:
any methods on ExecutionStrategy require thought bcz it is so general. even for things which may be on lots of APIs (like close) it may not always apply in the expected context (e.g. service filters are closable but don't need to be offloaded IIUC bcz they aren't invoked from the event loop but are driven by user code). If possible it would be nice to have a targeted approach on APIs (e.g. extending an influencer allows the user to control only what needs to be offloaded), however maybe we can make exceptions if this would overly complicate APIs/usage.